packages/components/src/inline-edit/hook.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { isUndefined, negate, noop, flow } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState, useRef } from '@wordpress/element';
+
+/**
+ * @param {import("react").ChangeEvent<HTMLInputElement>} event
+ */
+const cancelEvent = ( event ) => (
+	event.preventDefault(), event.stopPropagation(), event
+);
+
+/**
+ * @param {import("react").ChangeEvent<HTMLInputElement>} event
+ */
+const getEventValue = ( { target: { value } } ) => value;
+
+/**
+ * @param {Function[]} handlers
+ */
+const mergeEvent = ( ...handlers ) =>
+	/**
+	 * @param {import("react").ChangeEvent<HTMLInputElement>} event
+	 */
+	( event ) => handlers.forEach( ( handler = noop ) => handler( event ) );
+
+/**
+ * @typedef Props
+ * @property {(value: string) => boolean} validate predicate
+ * @property {( value: string ) => void} onWrongInput called when a validate predicate fails
+ * @property {( value: string ) => void} onCommit called on enter/blur
+ * @property {string} value input value
+ */
+
+/**
+ * @param {Props} props
+ */
+export default function useInlineEdit( {
+	validate = negate( isUndefined ),
+	onWrongInput = noop,
+	onCommit = noop,
+	value: propValue,
+} ) {
+	const [ isInEditMode, setIsInEditMode ] = useState( false );
+	const [ editingValue, setEditingValue ] = useState( propValue );
+	/** @type {import('react').RefObject<HTMLInputElement>} */
+	const inputRef = useRef( null );
+	/** @type {import('react').RefObject<HTMLInputElement>} */
```

Comment:
I guess it should be a button? cc @sarayourfriend 
```suggestion
	/** @type {import('react').RefObject<HTMLButtonElement>} */
```